_posts/2022-09-23-discrete-events-simulation-at-providence-outpatient-care-services-covid-19-recovery-scheduling/des-for-covid-19-recovery-scheduling.Rmd

```diff
@@ -0,0 +1,80 @@
+
+- The second part consists of logging, status update, capacity checking and calculation functions for the simulation process. Some of the process Detailed information of patient arrivals are calculated and recorded through these supporting functions.
+
+- The last part is the simulation model function, where it reads in user defined clinics’ capacity, registration time, and input data excel file. The model then reads the patient schedules and generates patients based on their scheduled appointment time. 
+
+
+### simulated outpatient process
+
+- INITIALIZATION: Patients have two types, virtual or in-person. Virtual patients can escape all the registration steps, directly start their therapy appointment if available and leave immediately after appointment. In-person patients would just simply follow the normal outpatient process. 
+- ARRIVAL: The simulated in-person patients would arrive to the registration desk, either on their own or by wheel-trans. Their arrival type is determined based on the wheel-trans arrival probability and python randomizer. In the registration wait room, if there is enough space to allow new patients in, new patients can then get registered; else new patients would have to wait for new capacity coming up to enter the registration room.
+- THERAPY: Once patients get registered, a schedule check would occur to see if the room and therapist booked for this patient(s) are available. If available, status would be updated, and patients can proceed to their appointments immediately. If not, patients would have to wait until the clinic/therapist is ready.
+- DIMISSAL: During dismissal, the arrival type is first checked. If the patient arrived by wheel-trans, then the registration wait room capacity would be checked to see if there is any space available for the leaving patients. If yes, wheel-trans patients can wait till their wheel-trans service arrive. If not, they will have to wait in the clinic wait rooms for an upcoming space. If the patient arrived on their own, they could leave immediately after their appointment. Status of current room capacities would be updated accordingly. 
+
+•	During this, process, all related/interested performance stats (wait time, current number in system) are recorded for result calculations. Utilization plots for clinics/therapists are also plotted for easier understanding. 
+Limitations:
+The simulation model itself basically replicates the normal outpatient service process but simplifies it by ignoring some of the variabilities in process times due to the lack of data. Instead of fitting a time distribution, we had to use a more deterministic approach by assuming a static service time for all the servicing times involved in the model. However, once enough data is collected, those distributions can be obtained by data-fitting and directly apply to the model. In this way, the model would have more randomness, and thus more realistic comparing to real life.
+
+## Evaluations
+
```

Comment:
Is the summary part of the evaluations section, or is there meant to be some additional content for this section?

Reply:
This was a placeholder since we did not have any evaluations.  The project was never used.